client/src/dashboard/ManageApplicationModal.js

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import axios from 'axios';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Grid from '@mui/material/Grid';
+import NiceModal, { useModal } from '@ebay/nice-modal-react';
+import TextField from '@mui/material/TextField';
+
+
+const ManageApplicationModal = NiceModal.create(({user_id, app=null}) => {
+    const modal = useModal();
+    // Set up states for an application
+    const isNewApplication = !app;
+    const [name, setName] = React.useState(isNewApplication ? "" : app.name);
+
+    const onNameChange = (event) => {
+        setName(event.target.value);
+    };
+
+    const onAddOrEdit = () => {
+        if (name != "") {
```

Comment:
Is it possible `name == null`?

Reply:
I'm not sure if users can name an new application with an empty name, so I just assume the name should be non-empty for now. Just to make sure, can the name be empty?